omigami/feature_evaluator.py

```diff
@@ -1,105 +1,54 @@
-        return self._n_features
-
-    def get_inner_loop_size(self):
-        return self._n_inner
-
-    def _make_metric(self, metric: Union[str, MetricFunction]) -> MetricFunction:
-        """Build metric function using the input `metric`. If a metric is a string
-        then is interpreted as a scikit-learn metric score, such as "accuracy".
-        Else, if should be a callable accepting two input arrays."""
-        if isinstance(metric, str):
-            return self._make_metric_from_string(metric)
-        elif hasattr(metric, "__call__"):
-            return metric
-        else:
-            raise ValueError("Input metric is not valid")
-
-    @staticmethod
-    def _make_metric_from_string(metric_string: str) -> MetricFunction:
-        if metric_string == "MISS":
-            return miss_score
-        if metric_string in SCORERS:
-            # pylint: disable=protected-access
-            return get_scorer(metric_string)._score_func
-        raise ValueError("Input metric is not a valid string")
-
-    def _get_feature_importances(self, estimator: Estimator):
-        if hasattr(estimator, "feature_importances_"):
-            return estimator.feature_importances_
-        elif hasattr(estimator, "coef_"):
-            return np.abs(estimator.coef_[0])
-        elif hasattr(estimator, "steps"):
-            for _, step in estimator.steps:
-                if hasattr(step, "coef_") or hasattr(step, "feature_importances_"):
-                    return self._get_feature_importances(step)
-        else:
-            raise ValueError("The estimator provided has no feature importances")
+        ranks = self._get_feature_ranks(estimator, features)
+        return FeatureEvaluationResults(test_score=score, ranks=ranks, model=estimator)
 
     def _get_feature_ranks(
-        self, estimator: Estimator, features: Iterable[int]
+        self, estimator: Estimator, features: Union[List[int], NumpyArray]
     ) -> FeatureRanks:
-        """Extract the feature rank from the input estimator. So far it can only handle
-        estimators as scikit-learn ones, so either having the `feature_importances_` or
-        the `coef_` attribute."""
-        feature_importances = self._get_feature_importances(estimator)
+        if self._n_initial_features == 0:
+            raise ValueError("Call set_n_initial_features first")
+        feature_importances = estimator.feature_importances
         ranks = rankdata(-feature_importances)
-        return FeatureRanks(features=features, ranks=ranks, n_feats=self._n_features)
-
-    def refresh_splits(self):
-        self._splitter = self._splitter.fit(self._input_data)
-
-
-def miss_score(y_true: NumpyArray, y_pred: NumpyArray):
-    """MISS score: number of wrong classifications preceded by - so that the higher
-    this score the better the model"""
-    return -(y_true != y_pred).sum()
+        return FeatureRanks(
+            features=features, ranks=ranks, n_feats=self._n_initial_features
```

Comment:
I'm guessing this is to be able to aggregate on the RFE level? I don't like it a lot here tbh..

Reply:
what exactly? not sure I understood what you meant

Reply:
I meant `_n_initial_features`